### PR TITLE
Camera and Scene integration (big)

### PR DESCRIPTION
**Pull request by elsien-dev**
- Adds the first-person camera to the Player
- Adds integration with Fly Camera so that you can switch between the two by playing/pausing the scene
- Camera Refactor: Camera is now separate from FlyCamera (may want to add option for fly controls instead, idk). FirstPersonCamera is FollowCameraController Component attached to the Player.
- Fixes the controls in the character controller to match the camera rotation.
- Adds SceneViews: This allows us to distinguish when we are viewing the scene from fly cam. versus. in game. It seems silly rn, but this is a pretty common thing to distinguish from what I've seen from other engines.

### CODE REVIEW

**Comment by keptsecret** [comment]
Does this include both #35 and #36?

**Comment by kevinsadi** [comment]
I know it includes #35 and I think it includes #36 as well, there's one commit where he said he added Ray's code. Also in the video he sent yesterday, it had Ray's code @keptsecret

File: src/engine/components/FollowCameraController.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+	void FollowCameraController::OnImGuiRender()
+	{
+	}
+
+	void FollowCameraController::Update()
+	{
+		//Modify the player's lookdir to match the camera look
+		m_FollowEntity->m_Transform.yawRotation = m_Camera->GetYaw();
+		m_Camera->SetPosition(m_FollowEntity->m_Transform.position + m_OffsetFromEntity);
```
Comment (kevinsadi) [inline src/engine/components/FollowCameraController.cpp:31]: good idea adding that offset as a member variable

File: src/engine/core/Camera.hpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			FIXED,
+		};
+
+	protected:
+		const float MAX_PITCH = 89.0f;
+		const float MOUSE_SENSITIVITY = 0.1f;
+
+		float m_Yaw;
+		float m_Pitch;
+		// We don't roll around here
```
Comment (kevinsadi) [inline src/engine/core/Camera.hpp:28]: that's just the way we roll 🤡

File: src/engine/core/SceneView.cpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		// load camera
+		GLFWwindow* const win = Application::Window();
+
+		// Add Default Fly Camera
+		m_FlyCamera = std::make_unique<FlyCamera>(win, 90.0f, 0.0f);
+	}
+
+	SceneView::~SceneView()
+	{
+		//lmfao
```
Comment (kevinsadi) [inline src/engine/core/SceneView.cpp:19]: LMFAO

File: src/engine/guitools/GameGUI.hpp [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <imgui.h>
+
+// This takes the place of the in-game gui for when we separate editor from game.
```
Comment (kevinsadi) [inline src/engine/guitools/GameGUI.hpp:5]: o7

**Review by kevinsadi — APPROVED** [review]
This is an insanely impressive PR. Good work my guy.

**Review by keptsecret — APPROVED** [review]

**Pull request merged**